sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -160,6 +160,15 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
       }
     }
   }
+
+  test("SPARK-23340 Empty float/double array columns raise EOFException") {
+    Seq(Seq(Array.empty[Float]).toDF(), Seq(Array.empty[Double]).toDF()).foreach { df =>
+      withTempPath { path =>
```

Comment:
Are we testing the native readers?

Reply:
Sure. This suite is in `sql/core` and inherits `OrcTest.scala`'s `val orcImp: String = "native"`.

Reply:
Please also test both readers as we discussed above.

Reply:
? I already added the test case at [HiveOrcQuerySuite.scala](https://github.com/apache/spark/pull/20511/files#diff-1569b2874975978ed62a01aab108d093R212), too.

Reply:
We have three ORC readers, right? We need to check all of them, and also vectorized reader too, even if they do not support it.

Reply:
Ur, I think you are still confused two things.

First of all, we have five ORC readers. We didn't check `ORC MR reader` and `ORC Vectorized Copy` explicitly. We usually test 1, 2, and 3.

1. Hive Serde
2. Hive OrcFileFormat
3. Apache ORC Vectorized Wrapper
4. Apache ORC Vectorized Copy
5. Apache ORC MR

In this PR, we already adds 1, 2, 3. 3 is the vectorized wrapper reader.
1. Hive Serde : `HiveOrcQuerySuite.test(SPARK-23340 Empty float/double array columns raise EOFException)`
2. Hive OrcFileFormat: `OrcSourceSuite` <= `OrcSuite.test("SPARK-23340 Empty float/double array columns raise EOFException")`
3. Apache ORC Vectorized Wrapper: `HiveOrcSourceSuite` <= `OrcSuite.test("SPARK-23340 Empty float/double array columns raise EOFException")`

Second, this test schema includes complex types. So, 3 (vectorized wrapper reader) configuration is also going to fall-back ORC MR Reader path. In other words, case 5. Please note that Apache Spark support vectorization for `Atomic` types only in both Parquet and ORC.

Reply:
When we write these test cases, we should do our best to cover all the scenarios even if it is not supported now. This is like black-box testing.

Reply:
It's not about ORC issue. Spark doesn't allow complex type vectorization. Please see `def supportBatch`.
> When we write these test cases, we should do our best to cover all the scenarios even if it is not supported now. This is like black-box testing.

Reply:
You know, we still can support complex type vectorization in the future, but we might not re-visist all the tests in the future.

Reply:
Then, since the current case is already testing against vectorized configuration *by default*.

We can add `Seq(false, true).foreach(... spark.sql.orc.enableVectorizedReader <- value)`. we can add `vectorization=false` for now. However, both test cases will test MR reader only for now. If it sounds okay to you, then I'll add that.